crates/bifrost/src/providers/replicated_loglet/sequencer/appender.rs

```diff
@@ -205,21 +206,18 @@ impl<T: TransportConnect> SequencerAppender<T> {
         ) {
             Ok(spread) => spread,
             Err(_) => {
-                if graylist.is_empty() {
-                    // gray list was empty during spread selection!
-                    // yet we couldn't find a spread. there is
-                    // no reason to retry immediately.
-                    return SequencerAppenderState::Backoff;
-                }
-                // otherwise, we retry without a gray list.
-                return SequencerAppenderState::Wave {
-                    graylist: NodeSet::empty(),
-                };
+                return SequencerAppenderState::Backoff;
```

Comment:
`Backoff` will retry the spread with an empty graylist anyway. So my reasoning was it's better to do that when selecting a spread fails regardless of if we have a spread or not

Reply:
Agreed